src/electron/process_manager.ts

```diff
@@ -334,53 +359,125 @@ class ChildProcessHelper {
     if (!this.process) {
       // Never started.
       if (this.exitListener) {
-        this.exitListener();
+        this.exitListener(null);
       }
       return;
     }
 
     this.process.kill();
   }
 
-  set onExit(newListener: (() => void)|undefined) {
+  set onExit(newListener: ((code: number) => void)|undefined) {
     this.exitListener = newListener;
   }
 
   /**
    * Enables verbose logging for the process.  Must be called before launch().
    */
-  public enableDebugMode() {
+  enableDebugMode() {
     this.isInDebugMode = true;
   }
 }
 
-class SsLocal extends ChildProcessHelper {
-  constructor(private readonly proxyPort: number) {
+// shadowsocks-libev client program. Starts a local SOCKS proxy that relays TCP/UDP
+// traffic to a Shadowsocks proxy server.
+class SsLocal extends ChildProcessHelper implements ShadowsocksProcess {
```

Comment:
The code for the two stacks should not live in the same file. Please move them to separate files so it's clear what's shared and what's not..